File: cesta/AMarotoChivite-cesta/AMarotoChivite-cesta-Java/src/test/java/simulacionCesta/models/FinalCestTest.java
package simulacionCesta.models;

import org.junit.jupiter.api.Test;

import java.time.LocalDate;

import static org.junit.jupiter.api.Assertions.assertEquals;


class FinalCestTest {

    @Test
    public void calculateTotalPriceTest() {
        LocalDate dateCreation = LocalDate.now();
        Product product1 = new Product("Agua", 1.0, 2);
        Product product2 = new Product("Pan", 2.0, 1);

        Product[] products = {product1, product2};
        CestList cestList = new CestList(2, products);
        FinalCest finalCest = new FinalCest(dateCreation, cestList);
        assertEquals(4.0, finalCest.calculateTotalPrice());
    }

    @Test
    void calculateQuantityProductsTest() {
        LocalDate dateCreation = LocalDate.now();
        Product product1 = new Product("Agua", 1.0, 2);
        Product product2 = new Product("Pan", 2.0, 1);

        Product[] products = {product1, product2};
        CestList cestList = new CestList(2, products);
        FinalCest finalCest = new FinalCest(dateCreation, cestList);
        assertEquals(3, finalCest.calculateQuantityProducts());
    }
}

File: cesta/AMarotoChivite-cesta/AMarotoChivite-cesta-Kotlin/src/test/kotlin/simulacionCesta/models/FinalCestTest.kt
package simulacionCesta.models

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import java.time.LocalDate

internal class FinalCestTest {
    @Test
    fun calculateTotalPriceTest() {
        val dateCreation = LocalDate.now()
        val product1 = Product("Agua", 1.0, 2)
        val product2 = Product("Pan", 2.0, 1)
        val products = arrayOf<Product?>(product1, product2)
        val cestList = CestList(2, products)
        val finalCest = FinalCest(dateCreation, cestList)
        Assertions.assertEquals(4.0, finalCest.calculateTotalPrice())
    }

    @Test
    fun calculateQuantityProductsTest() {
        val dateCreation = LocalDate.now()
        val product1 = Product("Agua", 1.0, 2)
        val product2 = Product("Pan", 2.0, 1)
        val products = arrayOf<Product?>(product1, product2)
        val cestList = CestList(2, products)
        val finalCest = FinalCest(dateCreation, cestList)
        Assertions.assertEquals(3, finalCest.calculateQuantityProducts())
    }
}